src/component/setup/props.js

```diff
@@ -16,44 +16,55 @@
  */
 
 import { Log } from '../../lib/log.js'
-
 import symbols from '../../lib/symbols.js'
 
-const baseProp = {
-  cast: (v) => v,
-  required: false,
+const normalizeProps = (props) => {
+  const out = {}
+
+  if (Array.isArray(props) === true) {
+    Log.warn(
+      'Defining props as an Array has been deprecated and will stop working in future versions. Please use the new notation instead (an object with key values pairs).'
+    )
+    const propLength = props.length
+    for (let i = 0; i < propLength; i++) {
+      const prop = props[i]
+      if (typeof prop === 'string') {
+        out[prop] = { default: undefined }
```

Comment:
I think these don't need the be storing objects with when you're only using 1 value in them.. Or am I missing something?

Reply:
this is for backwards compatibility - the old syntax could just be an array, in the new setup we're expecting an object. This piece of code rewrites it to the expected format